go/core/servers.go

```diff
@@ -43,36 +44,149 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-// StartFlowServer starts a server serving the routes described in [NewFlowServeMux].
-// It listens on addr, or if empty, the value of the PORT environment variable,
-// or if that is empty, ":3400".
-//
-// In development mode (if the environment variable GENKIT_ENV=dev), it also starts
-// a dev server.
-//
-// StartFlowServer always returns a non-nil error, the one returned by http.ListenAndServe.
-func StartFlowServer(addr string, flows []string) error {
-	return startProdServer(addr, flows)
-}
-
 // InternalInit is for use by the genkit package only.
 // It is not subject to compatibility guarantees.
-func InternalInit(opts *Options) error {
+func InternalInit(ctx context.Context, opts *Options) error {
 	if opts == nil {
 		opts = &Options{}
 	}
 	globalRegistry.freeze()
 
+	var servers []*http.Server
+	var wg sync.WaitGroup
+	errCh := make(chan error, 2)
+
 	if currentEnvironment() == EnvironmentDev {
+		wg.Add(1)
 		go func() {
-			err := startDevServer()
-			slog.Error("dev server stopped", "err", err)
+			defer wg.Done()
+			s, err := startReflectionServer(errCh)
+			if err != nil {
+				errCh <- fmt.Errorf("failed to start reflection server: %w", err)
+				return
+			}
+			servers = append(servers, s)
 		}()
 	}
-	if opts.FlowAddr == "-" {
-		return nil
+
+	if opts.FlowAddr != "-" {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			s, err := startFlowServer(opts.FlowAddr, opts.Flows, errCh)
+			if err != nil {
+				errCh <- fmt.Errorf("failed to start flow server: %w", err)
+				return
+			}
+			servers = append(servers, s)
```

Comment:
This reminded me that one of the error handlers won't ever do anything because we sent a permanent `nil` from `startServer`. Errors will only be sent on the channel now. PTAL.

Reply:
Sorry, I forgot about errgroup: https://pkg.go.dev/golang.org/x/sync@v0.7.0/errgroup
That will deal with errors and context cancellation.
If you want to move on, I'll approve this PR and use errgroup in my own PR.

Reply:
I gave it a shot refactoring with errgroup. The main err handling/waiting is cleaner, but it doesn't work without significant refactor of how `Server` is created/returned to be kept track of for shutdown since `ListenAndServe()` is a blocking function. I'll keep it as is for now and we can revisit in the future in needed.